### ISSUE TITLE
Destructuring and `prefer-const` Rule

### ISSUE THREAD

**Issue by jfftck** [palantir/tslint#3149]
### Bug Report

- __TSLint version__: 5.3.2
- __TypeScript version__: 2.4.2
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
    let [width, unit] = size.match(/^(\d*)(.*)$/).slice(1, 3)

    width = `${Number(width) * (this.wide ? 2 : 1)}${unit}`
```

with `tslint.json` configuration:

```json
{
  "prefer-const": true
}
```

#### Actual behavior
Does not pass the lint since unit could be initialized with `const`, but width must use `let` since it is being reassigned.

#### Expected behavior
Ignore this line since applying the "fix" would break the code.

**Comment by ajafff** [comment]
The rule already handles such cases with the appropriate configuration. From [the docs](https://palantir.github.io/tslint/rules/prefer-const/): `"prefer-const": [true, {"destructuring": "all"}]`
> “all” - Only warns if all variables in destructuring can be const.

**Comment by jfftck** [comment]
Is there a reason the configuration file does not default to the most sane values?

Also, I have found that documentation for the rules not always up-to-date.

**Comment by ajafff** [comment]
That's to keep the rules as strict as possible while still giving you the ability to opt out of certain checks. And maybe to keep it consistent with the behavior before that option was added.
Or I'm completely mistaken and we just forgot to enable the option back then when I added it...

Do you have some examples of outdated documentation? We can look into this and fix it.
